media/js/base/mozilla-client.js

```diff
@@ -0,0 +1,165 @@
+        ua = ua || navigator.userAgent;
+
+        var matches = /Firefox\/(\d+(?:\.\d+){1,2})/.exec(ua);
+
+        return (matches !== null && matches.length > 0) ? matches[1] : '0';
+    };
+
+    /**
+     * Detect whether the user's Firefox is up to date or outdated. This data is mainly used for security notifications.
+     *
+     * @param  {Boolean} strict - whether the minor and patch-level version numbers should be compared. Default: true
+     * @param  {Boolean} isESR - whether the Firefox update channel is ESR. Default: false
+     * @param  {String}  userVer - browser's version number
+     * @return {Boolean} result
+     */
+    Client._isFirefoxUpToDate = function (strict, isESR, userVer) {
+        strict = strict === undefined ? true : strict;
+        isESR = isESR === undefined ? false : isESR;
+        userVer = userVer === undefined ? this._getFirefoxVersion() : userVer;
+
+        var $html = $(document.documentElement);
+
+        if (!$html.attr('data-esr-versions') || !$html.attr('data-latest-firefox')) {
+            return false;
+        }
+
+        var versions = isESR ? $html.attr('data-esr-versions').split(' ') : [$html.attr('data-latest-firefox')];
+        var userVerArr = userVer.match(/^(\d+(?:\.\d+){1,2})/)[1].split('.');
+        var isUpToDate = false;
+
+        // Compare the newer version first
+        versions.sort(function(a, b) { return parseFloat(a) < parseFloat(b); });
+
+        // Only check the major version in non-strict comparison mode
+        if (!strict) {
+            userVerArr.length = 1;
+        }
+
+        for (var i = 0; i < versions.length; i++) {
+            var latestVerArr = versions[i].split('.');
+
+            // Only check the major version in non-strict comparison mode
+            if (!strict) {
+                latestVerArr.length = 1;
+            }
+
+            for (var j = 0; j < userVerArr.length; j++) {
+                if (Number(userVerArr[j]) < Number(latestVerArr[j] || 0)) {
+                    isUpToDate = false;
+                    break;
+                } else {
+                    isUpToDate = true;
+                }
+            }
+
+            if (isUpToDate) {
+                break;
+            }
+        }
+
+        return isUpToDate;
+    };
+
+    /**
+     * Use the async mozUITour API of Firefox to retrieve the user's browser info, including the update channel and
+     * accurate, patch-level version number. This API is available on Firefox 35 and later. See
+     * http://bedrock.readthedocs.org/en/latest/uitour.html for details.
+     *
+     * @param  {Function} callback - callback function to be executed with the Firefox details
+     * @return {None}
+     */
+    Client.getFirefoxDetails = function (callback) {
+        // Fire the callback function immediately if cache exists
+        if (this.FirefoxDetails) {
+            callback(this.FirefoxDetails);
+
+            return;
+        }
+
+        var self = this;
```

Comment:
Seeing as this is a static object, is there any need to use `this`, and not reference `Client` explicitly?